posthog/api/project.py

```diff
@@ -424,6 +425,18 @@ def get_serializer_class(self) -> type[serializers.BaseSerializer]:
             return ProjectBackwardCompatBasicSerializer
         return super().get_serializer_class()
 
+    def dangerously_get_required_scopes(self, request, view) -> list[str]:
+        # If the request only contains config fields, require read:team scope
+        # Otherwise, require write:team scope (handled by APIScopePermission)
+        if self.action == "partial_update":
+            request_fields = set(request.data.keys())
+            non_config_fields = request_fields - set(CONFIG_FIELDS)
+            if not non_config_fields:
+                return ["team:read"]
+
+        # Fall back to the default behavior
+        return None
```

Comment:
For the /projects/ endpoint, we should be overriding things a little differently for `partial_update`: if the payload contains a Team config field, we should be ADDING `team:write`, to the base `project:read` and `project:write`.

Alternatively, the logic can be the same, with the override returning `["project:read"]`

Reply:
Good call, will update.